sol/main.c

```diff
@@ -617,31 +690,29 @@ run_sol(struct net_config *net_conf, struct sol_config *sol_conf)
 	 * Although the packets are going to the back interface,
 	 * they are allocated at the front interface.
 	 */
-	front_inc = 2 * sol_conf->pri_req_max_len + sol_conf->enq_burst_size;
+	front_inc = (2 * sol_conf->pri_req_max_len +
+		sol_conf->enq_burst_size) * sol_conf->num_lcores;
 	net_conf->front.total_pkt_burst += front_inc;
 
-	sol_conf->ring = rte_ring_create("sol_reqs_ring",
-		rte_align32pow2(sol_conf->pri_req_max_len),
-		rte_lcore_to_socket_id(sol_conf->lcore_id), RING_F_SC_DEQ);
-	if (sol_conf->ring == NULL) {
-		G_LOG(ERR,
-			"mailbox: can't create ring sol_reqs_ring at lcore %u\n",
-			sol_conf->lcore_id);
-		ret = -1;
-		goto burst;
-	}
+	if (sol_conf->num_lcores <= 0)
+		goto out;
```

Comment:
These lines must come earlier in order to use the label `out`. Here, it would require the label `burst`.

Reply:
Why not moving this test before the burst section and keep the label `out`? the code reads simpler this way.

Reply:
Agreed. Done.